app/models/concerns/parameterizable.rb

```diff
@@ -28,7 +36,15 @@ def to_param
       end
 
       def self.from_param(id_name)
-        self.find(id_name.to_i)
+        self.where(id: id_name.to_i).first
+      end
+
+      def self.find(*args)
+        if args.size == 1 && (args.first.is_a?(String) || args.first.is_a?(Numeric))
```

Comment:
Can we refactor all the .find methods to just one? They only differ (slightly!) in the conditional.